### PR TITLE
feat: P4ADEV-320-refactoring-accordingly-handbook

### PR DESCRIPTION
**Pull request by LarissaASLeite**
#### How Has This Been Tested?

- Pre-Deploy Test
    - [ ] Unit
    - [ ] Integration (Narrow)
- Post-Deploy Test
    - [ ] Isolated Microservice
    - [ ] Broader Integration
    - [ ] Acceptance
    - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [ ] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java [hunk 1]
```diff
@@ -40,7 +54,7 @@ public Map<String, String> validate(String token, String urlJwkProvider) {
         } catch (com.auth0.jwt.exceptions.TokenExpiredException e){
             throw new TokenExpiredException(e.getMessage());
         } catch (JwkException | JWTVerificationException ex) {
-            throw new InvalidTokenException(AuthConstants.ExceptionCode.INVALID_TOKEN, "The token is not valid", true, ex);
+            throw new InvalidTokenException(AuthErrorDTO.CodeEnum.INVALID_TOKEN.getValue(), "The token is not valid", true, ex);
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java:57]: the first param is an openapi implementation detail that should not be visible in the service layer

**Review by and-mora — CHANGES_REQUESTED** [review]

**Comment by and-mora** [comment]
do we plan to refactor the `logback-spring.xml` file?

**Comment by and-mora** [comment]
is the `MemoryAppender.java` class needed?

File: src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java (hunk 1 above)
Comment (LarissaASLeite) [reply src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java:57]: The code we are using here is not in any way part of HTTPstatus,  is just a business code, not a presentation layer. If we put a string we can risk to put some inexistent Enum

**Comment by LarissaASLeite** [comment]
> is the `MemoryAppender.java` class needed?

File: src/main/java/it/gov/pagopa/payhub/auth/exception/AuthExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public AuthErrorDTO handleInvalidTokenException(ServiceException ex, HttpServletRequest request){
+        logException(ex, request);
+        String message = ex.getMessage();
+
+
+        return new AuthErrorDTO(ex.getCode(), message);
+    }
+
+    public static void logException(ServiceException error, HttpServletRequest request) {
+        if(error.isPrintStackTrace()){
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/payhub/auth/exception/AuthExceptionHandler.java:31]: in which case we want to print a stack trace? `InvalidTokenException` and `TokenExpiredException` are very precise business exceptions which i expect already logs something when they're triggered. In this case what's the added value of a stack trace?
For observability purpose the exception will be visible and inspectable on ApplicationInsight

File: src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java (hunk 1 above)
Comment (and-mora) [reply src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java]: sorry, let me rephrase: do we need to put that string/enum there? Isn't that code already codified in the exception itself? 
e.g. `InvalidTokenException` codifies a `INVALID_TOKEN` code. Why we explicitly put that code in the throw?

File: .devops/security-scan-pipelines.yml
Comment (and-mora) [inline .devops/security-scan-pipelines.yml:1]: let's convert the security scan in a github action (like idpay)

Comment (and-mora) [reply .devops/security-scan-pipelines.yml:1]: actually that one in idpay repo's need to be changed to support protected environments. I would move the implementation of the security scan out of this PR.
